backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/ConfirmationCodeController.java

```diff
@@ -0,0 +1,48 @@
+package ca.gov.dtsstn.cdcp.api.web.v1.controller;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ca.gov.dtsstn.cdcp.api.service.ConfirmationCodeService;
+import ca.gov.dtsstn.cdcp.api.web.v1.model.ConfirmationCodeModel;
+import ca.gov.dtsstn.cdcp.api.web.v1.model.mapper.ConfirmationCodeModelAssembler;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.constraints.NotBlank;
+import ca.gov.dtsstn.cdcp.api.service.domain.ConfirmationCode;
+
+@Validated
+@RestController
+@RequestMapping({ "/api/v1" })
+@Tag(name = "confirmationCodes", description = "CRUD endpoint for confirmation codes.")
+public class ConfirmationCodeController {
+    
+    private final ConfirmationCodeModelAssembler confirmationCodeModelAssembler;
+    private final ConfirmationCodeService confirmationCodeService;
+
+    public ConfirmationCodeController(ConfirmationCodeModelAssembler confirmationCodeModelAssembler, ConfirmationCodeService confirmationCodeService){
+        this.confirmationCodeModelAssembler = confirmationCodeModelAssembler;
+        this.confirmationCodeService = confirmationCodeService;
+    }
+
+    @GetMapping({ "/users/{userEmail}/confirmationCodes" })
```

Comment:
- Anything starting with `/users` should probably go in the users controller.
- It should be `/users/{userId}/confirmation-codes`

Reply:
how about /confirmation-codes/{userEmail} ??

Reply:
`/confirmation-codes/{email}` isn't really a RESTful URI. Since fetching by email address would be considered a search, the parameter should go in the query string.. maybe `/confirmation-codes?filter=email:user@example.com` or something similar.

It's a bit of a moot point, since I don't think we need a `ConfirmationCodesController`, anyway. When the user logs in, we will have the user id.. so we can just fetch all of the user's confirmation codes through the `/users/{userId}/confirmation-codes` resource.